blob/service.go

```diff
@@ -167,22 +170,17 @@ func (s *Service) Subscribe(ctx context.Context, ns libshare.Namespace) (<-chan
 // Allows sending multiple Blobs atomically synchronously.
 // Uses default wallet registered on the Node.
 // Handles gas estimation and fee calculation.
-func (s *Service) Submit(ctx context.Context, blobs []*Blob, txConfig *SubmitOptions) (uint64, error) {
-	log.Debugw("submitting blobs", "amount", len(blobs))
-
+func (s *Service) Submit(ctx context.Context, blobs []*Blob, txConfig *SubmitOptions) (_ uint64, err error) {
 	libBlobs := make([]*libshare.Blob, len(blobs))
 	for i := range blobs {
-		if err := blobs[i].Namespace().ValidateForBlob(); err != nil {
-			return 0, fmt.Errorf("not allowed namespace %s were used to build the blob", blobs[i].Namespace().ID())
-		}
```

Comment:
I moved this level down to state.submit. I dont know why we did this check only here, but have a feeling that there could have been something. Will revert if we find the reason